doc/userguide/rules/flow-keywords.rst

```diff
@@ -318,6 +318,34 @@ Signature example::
 
 In this example, we combine `flow.age` and `flowbits` to get an alert on the first packet after the flow's age is older than one hour.
 
+flow.rate
+---------
+
+The rate of the flow calculated by dividing number of bytes by the time in seconds.
+So, the unit of the flow rate is bytes/second. Note that it is possible to denote
+the rate with units like kb, mb, etc.
+Currently, it is implemented as a check against the total number of bytes seen by
+the flow in both the directions divided by the age of the flow.
+
+For example, if a flow has seen 5000000 bytes in 10 seconds, a rule for this flow
+will be matched against 500000 bytes/s.
+
+flow.rate uses an :ref:`unsigned 64-bit integer <rules-integer-keywords>`.
+
+Syntax::
+
+ flow.rate: [op]<number>
+
+The rate can be matched exactly, or compared using the _op_ setting::
+
+ flow.rate:10000    # exactly 10000 bytes per second
+ flow.rate:>20000   # greater than 20000 bytes per second
+ flow.rate:>=30mb   # greater than equal to 30mbps
+
+Signature example::
+
+ pass tcp any any -> any any (msg:"Flow rate higher than 50kbps"; flow.rate:>50kb; sid:1; rev:1;)
```

Comment:
Not a fan of mismatching units. Considering keeping the initial syntax something like `flow.rate:>500kb,10` where 50kb is the number of bytes and 1 the seconds with which the rate will be calculated.

Reply:
Some other ideas, none of which I'm a fan of:
```
flow.rate: 50kbps
flow.rate: 50kb,10  # document that first arg is bytes and second one the number of seconds
flow.rate: 50kb,10
flow.rate: 50kb  # document that this means 50kbytes per second
flow.rate: bytes 50kb, seconds 10  # inspired by threshold keyword
```

Reply:
Between those, I like:
```
flow.rate: 50kb,10s
```
The most, I think. I imagine that any format would need documenting.

Reply:
You may also want packets instead of bytes like `flow.rate: packets 50, seconds 10`

Reply:
Will you also want milliseconds ?

Reply:
> Will you also want milliseconds ?

@catenacyber I don't know yet. Do you think somebody might want to express rate as 10k bytes in 5ms?

Reply:
I do not know either, but maybe more granularity/expressivity is always better ?

Reply:
Agreed. Shall keep that in mind. Thanks a lot for your valuable feedback, both! 🙇🏽‍♀️